### PR TITLE
feat: develop events page

### PR DESCRIPTION
**Pull request by nikvoblikov**
**This PR:**
Adds an Events page:
- Updated "Filters" and "Pagination" components for use in "Events" and "Blog";
- Adds links to the footer and menu "News and Media";
- Adds necessary events.

TODO before merge:
- [x]  Delete test events;

**Steps to test**

- Open the [Events page](https://deploy-preview-279--cilium.netlify.app/events/)
- Open the [Blog page](https://deploy-preview-279--cilium.netlify.app/blog/)
- Confirm that everything looks and works as expected

### CODE REVIEW

File: gatsby-node.js [hunk 1]
```diff
@@ -142,9 +144,122 @@ async function createBlogPages({ graphql, actions, reporter }) {
...
+  function getFrontmatterData(items) {
+    return items.map((item) => ({ ...item.frontmatter }));
+  }
+
+  const postEvents = getFrontmatterData(allEvents);
+  const featuredEvent = featuredPost[0].frontmatter;
+  const types = allTypes.reduce((acc, type) => [...acc, type.fieldValue], []);
+  const regions = allRegions.reduce((acc, type) => [...acc, type.fieldValue], []);
+  const eventFilters = createEventFilters(types, regions);
+  const initialFilters = getInitialFilters(eventFilters);
```
Comment (vannyle) [inline gatsby-node.js:244]:
```suggestion
  const initialFilters = allFilters.reduce((acc, { label }) => {
    if (!acc[label]) {
      acc[label] = [];
    }
    return acc;
  }, {});
```

File: src/utils/event-filters.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+}
+
+function getInitialFilters(allFilters) {
+  return allFilters.reduce((acc, { label }) => {
+    if (!acc[label]) {
+      acc[label] = [];
+    }
+    return acc;
+  }, {});
+}
```
Comment (vannyle) [inline src/utils/event-filters.js:27]:
```suggestion

```

File: src/utils/event-filters.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+// We are using ES modules here in order to be able to import variables from this file in gatsby-node.js
+module.exports = {
+  createEventFilters,
+  getInitialFilters,
```
Comment (vannyle) [inline src/utils/event-filters.js:32]:
```suggestion
```

File: gatsby-node.js (hunk 1 above)
Comment (vannyle) [reply gatsby-node.js:244]: IMHO, as it was not a reusable function, it is a shorter way to just get initialFilters here, not importing from utils.

**Review by vannyle — CHANGES_REQUESTED** [review]

File: src/components/pages/events/events-board/event-card/event-card.jsx [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+  <Link
+    to={externalUrl}
+    className={classNames(
+      'group flex flex-col rounded-lg border-2 border-gray-6 transition-all duration-200 hover:border-transparent hover:shadow-tertiary',
+      className
+    )}
+    target="_blank"
+    rel="noopener noreferrer"
+  >
+    <EventCover ogImage={ogImage} title={title} />
```
[the remark this answers is not in the document]
Comment (nikvoblikov) [reply src/components/pages/events/events-board/event-card/event-card.jsx:62]: This component is in /events/events-board/event-card/ that's why it's called that:)

File: src/components/pages/events/events-board/event-card/event-card.jsx [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+      >
+        <img className="h-full w-full py-2.5" src={placeholderIllustration} alt="" aria-hidden />
+      </div>
+    );
+  }
+
+  return content;
+};
+
+const BlogPostCard = ({
```
Comment (vannyle) [inline src/components/pages/events/events-board/event-card/event-card.jsx:43]: The component BlogPostCard?

**Review by sashasushko — APPROVED** [review]
LGTM 🔥 

Only one thing... please hide "[What’s New with Cilium & eBPF South Bay Meetup](https://isovalent.com/whats-new-with-cilium-ebpf-south-bay-meetup/)" because it give 404

**Comment by xmulligan** [comment]
Great, can we make sure all of the events have thumbnails not just the generic one and can we make this CiliumCon the featured one https://events.linuxfoundation.org/kubecon-cloudnativecon-north-america/co-located-events/ciliumcon/ and move the other one down?

File: content/events/2023-04-18-ciliumcon/index.md [hunk 6]
```diff
@@ -0,0 +1,11 @@
+---
+date: '2023-04-18T11:00:00.000Z'
+region: EMEA
+place: Amsterdam, The Netherlands
+title: 'CiliumCon'
```
Comment (xmulligan) [inline content/events/2023-04-18-ciliumcon/index.md:5]:
```suggestion
title: 'CiliumCon EU 2023'
```

File: content/events/2023-11-06-ciliumcon-na/index.md [hunk 7]
```diff
@@ -0,0 +1,11 @@
+---
+date: '2023-11-06T11:00:00.000Z'
+region: North America
+place: Chicago, Illinois
+title: 'CiliumCon'
```
Comment (xmulligan) [inline content/events/2023-11-06-ciliumcon-na/index.md:5]:
```suggestion
title: 'CiliumCon NA 2023'
```

**Comment by xmulligan** [comment]
Sorry I wasn't more clear. What I meant is that I wanted the image listing all of the talks as the thumbnail on the events page, not the other way around

**Comment by nikvoblikov** [comment]
I have changed the pictures. I want to note that for correct display for events, images must be 760x360 pixels in size, so I slightly changed the aspect ratio of the image.

**Comment by nikvoblikov** [comment]
> Sorry I wasn't more clear. What I meant is that I wanted the image listing all of the talks as the thumbnail on the events page, not the other way around

I have changed the pictures. I want to note that for correct display for events, images must be 760x360 pixels in size, so I slightly changed the aspect ratio of the image.

**Review by xmulligan — APPROVED** [review]

**Pull request merged**